nipype/algorithms/confounds.py

```diff
@@ -106,16 +270,121 @@ def _run_interface(self, runtime):
+      from nitime
+      <http://nipy.org/nitime/api/generated/nitime.algorithms.autoregressive.html\
+#nitime.algorithms.autoregressive.AR_est_YW>`_
+      for the :abbr:`AR (auto-regressive)` filtering of the fMRI signal.
+
+    :param numpy.ndarray func: functional data, after head-motion-correction.
+    :param numpy.ndarray mask: a 3D mask of the brain
+    :param bool output_all: write out all dvars
+    :param str out_file: a path to which the standardized dvars should be saved.
+    :return: the standardized DVARS
+
+    """
+    import os.path as op
+    import numpy as np
+    import nibabel as nb
+    from nitime.algorithms import AR_est_YW
+
+    func = nb.load(in_file).get_data().astype(np.float32)
+    mask = nb.load(in_mask).get_data().astype(np.uint8)
+
+    if len(func.shape) != 4:
+        raise RuntimeError(
+            "Input fMRI dataset should be 4-dimensional")
+
+    # Robust standard deviation
+    func_sd = (np.percentile(func, 75, axis=3) -
+               np.percentile(func, 25, axis=3)) / 1.349
+    func_sd[mask <= 0] = 0
+
+    # ar1_img = np.zeros_like(func_sd)
+    # ar1_img[idx] = diff_SDhat
+    nb.Nifti1Image(func_sd, nb.load(in_mask).get_affine()).to_filename('func_sd.nii.gz')
+
+
+    if remove_zerovariance:
+        # Remove zero-variance voxels across time axis
+        mask = zero_variance(func, mask)
+
+    idx = np.where(mask > 0)
+    mfunc = func[idx[0], idx[1], idx[2], :]
+
+    # Demean
+    mfunc -= mfunc.mean(axis=1).astype(np.float32)[..., np.newaxis]
+
+    # Compute (non-robust) estimate of lag-1 autocorrelation
+    ar1 = np.apply_along_axis(AR_est_YW, 1, mfunc, 1)[:, 0]
+
+    # Compute (predicted) standard deviation of temporal difference time series
+    diff_SDhat = np.squeeze(np.sqrt(((1 - ar1) * 2).tolist())) * func_sd[mask > 0].reshape(-1)
+    diff_sd_mean = diff_SDhat.mean()
+
+    # Compute temporal difference time series
+    func_diff = np.diff(mfunc, axis=1)
+
+    # DVARS (no standardization)
+    dvars_nstd = func_diff.std(axis=0)
+
+    # standardization
+    dvars_stdz = dvars_nstd / diff_sd_mean
+
+    # voxelwise standardization
+    diff_vx_stdz = func_diff / np.array([diff_SDhat] * func_diff.shape[-1]).T
+    dvars_vx_stdz = diff_vx_stdz.std(axis=0, ddof=1)
+
+    return (dvars_stdz, dvars_nstd, dvars_vx_stdz)
+
+def zero_variance(func, mask):
+    """
+    Mask out voxels with zero variance across t-axis
+
+    :param numpy.ndarray func: input fMRI dataset, after motion correction
+    :param numpy.ndarray mask: 3D brain mask
+    :return: the 3D mask of voxels with nonzero variance across :math:`t`.
+    :rtype: numpy.ndarray
+
+    """
+    idx = np.where(mask > 0)
+    func = func[idx[0], idx[1], idx[2], :]
+    tvariance = func.var(axis=1)
+    tv_mask = np.zeros_like(tvariance, dtype=np.uint8)
+    tv_mask[tvariance > 0] = 1
+
+    newmask = np.zeros_like(mask, dtype=np.uint8)
+    newmask[idx] = tv_mask
+    return newmask
```

Comment:
So you mean, mask out all zero-variance pixels? There is this little function to do that now :)